src/VecSim/spaces/spaces.cpp

```diff
@@ -53,25 +53,29 @@ CalculationGuideline FP64_GetCalculationGuideline(size_t dim) {
 
 void SetDistFunc(VecSimMetric metric, size_t dim, dist_func_t<float> *index_dist_func) {
 
+    static const Arch_Optimization arch_opt = getArchitectureOptimization();
```

Comment:
will this work for multiple indexes?

Reply:
yes. the function will be called only once (for each type) and then will use the `arch_opt` as a constant